byterun/fiber.c

```diff
@@ -73,6 +74,10 @@ static value save_stack ()
   Assert(caml_stack_high == Stack_high(old_stack));
   Assert(caml_extern_sp == caml_stack_high + Stack_sp(old_stack));
   dirty_stack(old_stack);
+  if (!Is_minor(old_stack) && caml_is_marked(old_stack)) {
+    //TODO: Only scan mutated part of the stack
+    caml_scan_stack(&caml_darken, old_stack);
+  }
```

Comment:
I don't understand why this is necessary. Shouldn't the call to `dirty_stack` above suffice? The minor GC scans all dirty stacks and calls caml_darken, regardless of whether they have been marked (see `clean_stacks` in `minor_gc.c`).